### PR TITLE
Log exception when something unexpected goes wrong during the VIES Va…

### LINKED ISSUE #36065: When VAT number validation fails by the VIES service, there is no trace about the exception message in the log files
### Preconditions and environment

- Magento version: any

### Steps to reproduce

Look at the code over here: https://github.com/magento/magento2/blob/2.4.5/app/code/Magento/Customer/Model/Vat.php#L214-L218

1. In the code [over here](https://github.com/magento/magento2/blob/2.4.5/app/code/Magento/Customer/Model/Vat.php#L47), change the url to https://example.com/taxation_customs/vies/checkVatService.wsdl so we can simulate an unexpected error
2. In the Magento back office, go to Stores > Configuration > General > General > Store Information
3. For the Country field, pick an EU country (like France)
4. For the VAT Number field, enter something random, like 012345
5. Press the Validate VAT Number button
6. Inspect the var/log/exception.log file or system.log file

### Expected result

1. Expect that the exception message is logged somewhere, so that when an exception is occurring, it's easy to find the details in a log file

### Actual result

1. No exception message is being logged

### Additional information

We had a certain shop, where the VAT number validation kept on failing and there was no trace about why it happened, when we tried debugging on our local environment, no issue was found, it only happened on the staging server.

After some debugging directly on the staging server, we figured out that an exception was being thrown by that SoapClient and it was being [catched](https://github.com/magento/magento2/blob/2.4.5/app/code/Magento/Customer/Model/Vat.php#L214-L218), but not logged:
```
javax.xml.rpc.soap.SOAPFaultException: IP_BLOCKED [] []
```

If we would have seen this in some log file, it would have been easier to figure out exactly why the vat number validation failed in our case.

### Release note

Exceptions thrown by the [VIES VAT validation service](https://ec.europa.eu/taxation_customs/vies/#/vat-validation) are now being logged in the `var/log/exception.log` file so it becomes easier to figure out what goes wrong when the validation doesn't work.

### Triage and priority

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

### PR DESCRIPTION
**Pull request by hostep**
…t Number validation process.

### Description (*)
Whenever something unexpected goes wrong with the [VIES VAT number validation service](https://ec.europa.eu/taxation_customs/vies/#/vat-validation) (for example: it being down, or having changed url's, or your server being blocked because having send too many requests, ...) no trace of this was being logged. Which made it hard to figure out exactly what the problem is without having to debug things manually.
This PR changes that and makes sure the exception information gets logged in `var/log/exception.log`

### Fixed Issues (if relevant)
1. Fixes https://github.com/magento/magento2/issues/36065

### Manual testing scenarios (*)
1. In the code [over here](https://github.com/magento/magento2/blob/2.4.5/app/code/Magento/Customer/Model/Vat.php#L47), change the url to `https://example.com/taxation_customs/vies/checkVatService.wsdl` so we can simulate an unexpected error
2. In the Magento backoffice, go to Stores > Configuration > General > General > Store Information
3. For the Country field, pick an EU country (like France)
4. For the VAT Number field, enter something random, like `012345`
5. Press the `Validate VAT Number` button
6. Inspect the `var/log/exception.log` file
7. With this PR applied, it's expected that an entry is created there to give some hints about what goes wrong with the VAT Number validation service 

### Questions or comments
I haven't added unit tests or any other test, there seems to be very little automatic tests that exist for this part in Magento and I don't want to spend the time in to write everything out from scratch, I don't have the energy or time for that, so if this is required, it would be appreciated if an Adobe employee could help out with that.

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] README.md files for modified modules are updated and included in the pull request if any [README.md predefined sections](https://github.com/magento/devdocs/wiki/Magento-module-README.md) require an update
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by hostep** [comment]
@magento run all tests

File: app/code/Magento/Customer/Model/Vat.php [hunk 1]
```diff
@@ -212,6 +212,11 @@ public function checkVatNumber($countryCode, $vatNumber, $requesterCountryCode =
                 $gatewayResponse->setRequestMessage(__('Please enter a valid VAT number.'));
             }
         } catch (\Exception $exception) {
+            $this->logger->error(
+                __('VAT Number validation failed with message: %1', $exception->getMessage()),
```
Comment (kandy) [inline app/code/Magento/Customer/Model/Vat.php]: Logged data does not need to be translated

Comment (hostep) [reply app/code/Magento/Customer/Model/Vat.php]: Makes sense to me (I got incorrectly inspired by some of the other translated logged data in Magento's codebase).

Anyway, fixed now!

**Comment by hostep** [comment]
@magento run all tests

**Comment by hostep** [comment]
Failed tests don't seem to be related to this PR.

**Comment by hostep** [comment]
Hey @kandy: what needs to happen to move this PR forwards? It looks like it's stuck for some unknown reason?

**Review by kandy — APPROVED** [review]

**Comment by engcom-Lima** [comment]
@magento run all tests

**Comment by engcom-Lima** [comment]
:heavy_check_mark: QA Passed

**Preconditions:**
- Install fresh Magento **`2.4-develop`** and **`PHP 8.1`**

**Manual testing scenario:**

1. In the code [over here](https://github.com/magento/magento2/blob/2.4.5/app/code/Magento/Customer/Model/Vat.php#L47), change the url to https://example.com/taxation_customs/vies/checkVatService.wsdl so we can simulate an unexpected error
2. In the Magento backoffice, go to Stores > Configuration > General > General > Store Information
3. For the Country field, pick an EU country (like France)
4. For the VAT Number field, enter something random, like 012345
5. Press the Validate VAT Number button
6. Inspect the var/log/exception.log file
7. With this PR applied, it's expected that an entry is created there to give some hints about what goes wrong with the VAT Number validation service

**Before:** :heavy_multiplication_x:  No exception message is being logged


**After:** :heavy_check_mark: Now it is coming.
Screenshot 2023-04-14 at 1 19 17 PM


Builds are failed. Hence, moving this PR to **Extended Testing.**

**Comment by engcom-Lima** [comment]
@magento run all tests

**Comment by engcom-Lima** [comment]
@magento run Functional Tests B2B

**Comment by engcom-Lima** [comment]
@magento run Functional Tests B2B

**Comment by engcom-Charlie** [comment]
@magento run Functional Tests CE, Functional Tests EE

**Comment by engcom-Charlie** [comment]
@magento run Functional Tests CE, Functional Tests EE

**Comment by engcom-Charlie** [comment]
The functional CE failures seems to be flaky as they are different in recent to builds.

Build 1:
image


Build 2:
image

**Comment by engcom-Charlie** [comment]
@magento run  Functional Tests EE

**Comment by engcom-Charlie** [comment]
@magento run Functional Tests EE

**Comment by engcom-Charlie** [comment]
The Functional EE failures are different in latest 3 run on the same commit so considering them as flaky and not related to this PR.

**Functional EE**:
**Run 1**:
image

**Run 2**:
image

**Run 3**:
image


Hence moving this PR in Merge in Progress now.

Thank you!

**Comment by engcom-Charlie** [comment]
@magento run all tests

**Comment by hostep** [comment]
It looks like this PR got merged in 2.4-develop recently via: https://github.com/magento/magento2/commit/d4a9791d0f589ee1006beba25c240a39a59d50e9

But this PR is not marked as merged, which is quite strange? Should I close it myself?

**Comment by ihor-sviziev** [comment]
Changes from this PR were merged in https://github.com/magento/magento2/commit/50b303edae8c26ceb24fc23dcb42b1ed2feb1b76.
Unfortunately, the last merge commit wasn't included, so Github isn't marked PR as "merged" :(

**Pull request closed without merging**